lib/filewatch/watch.rb

```diff
@@ -118,13 +118,21 @@ def each(&block)
       synchronized do
         return if @files.empty?
 
+        quitting = false
         file_deleteable = []
         # creates this array just once
         watched_files = @files.values
 
+        quit_proc = lambda do
+          quitting.tap do |flag|
+            file_deleteable.each {|f| @files.delete(f)} if flag
```

Comment:
oh I see, further down you use quit_proc as a "should we quit, and also do cleanup if we are quitting" Kind of thing. right?